Frontend search UI

alexandercyu opened the pull request:
- Details about a particular search result are displayed as depicted in the FIgma design, with some changes as discussed in meetings
   -> removing BBox section, adding a map to display BBox and geometry, displaying metadata in a nested fashion, etc.

- Filter functions added for Catalog dropdown, and partially to the Time Frame selection.

Search results for the Catalog filter seem correct.
Search results for the Time Frame filter don't seem correct.

Edit 2025-06-18:
- Time frame panel repeat functions removed for now
- search box retooled so it adds keywords to the Other section instead of being a search on its own
- added dropdowns on the left side categorized according to extension
- dropdowns on left side will get data from the "collections" endpoint if the "queryables" endpoint doesn't exist

alexandercyu committed:
Update dropdown-panel.html

- add bootstrap close attribute so the dropdown collapses when clicking outside of it

alexandercyu committed:
Update index.html

- remove class from dataset row
- remove title container and title tags from index

alexandercyu committed:
Update catalog-search.js

- add function to initialize bootstrap tooltips
- remove duplicated condition featureKey == "assets"
- move featureKey == "properties" condition to below "assets"
- add tooltips to links and icons that are supposed to copy the link to the clipboard
- add css classes to icons, icon container
- add function setClipboard() to allow writing the dataset link to the clipboard on click
- add function hideTooltip() to  hide the tooltip after 2 seconds

alexandercyu committed:
Update catalog-search.js

Remove unneeded parameter

alexandercyu committed:
Update stac-ui.css

- Added radial gradient hover effect on hover of download icon as defined in design
- added additional width attributes for asset links
-added styling for tooltips (Not working)

alexandercyu committed:
Update catalog-search.js

- removed commented code
- add "properties" object definition to JSON that will be sent in search query
- add code to dynamically add collection name to search results
- change condition for code that displays assets to use current stac collection json from redoak and not the testing server
- add hover tooltips to let user know to click links and icons to copy the url link or download the file
- add Variables and Dimensions sections
- add function to build the header template of each section
- add testing function that passes the string representation of the stac json from redoak

alexandercyu committed:
Update stac-ui.css

- add styles for Variables and Dimensions sections

alexandercyu committed:
Update index.html

- Add section for collection name
- remove spaces from html tags

alexandercyu committed:
Update catalog-search.js

- Added production version of populateSearchResults() function that creates links from the live redoak stac search json
- removed function that makes table header for the default search results when the page first loads
- Add code to build Metadata section

alexandercyu committed:
Update stac-ui.css

Add classes for metadata section header

alexandercyu committed:
Update catalog-search.js

-Add header for node information section
- add function to explicitly hide the hover tooltip and show the click tooltip when clicking on a link or clicking on the copy icon
- Add comments to denote different sections of the page they populate
- remove commented code
- add Marble section
- add function to show tooltip

alexandercyu committed:
Update stac-ui.css

- add styles for node section

alexandercyu committed:
Update leaflet-ui.css

- change map styles to a class instead of targeting an ID

alexandercyu committed:
Update stac-ui.css

- move bottom border styles to their own classes
- add border styles for ul lists
- add border styles for div lists
- add border styles for tables

alexandercyu committed:
Update index.html

- add container for leaflet map
- add leaflet function files

alexandercyu committed:
Update catalog-search.js

- add classes for borders to sections

alexandercyu committed:
Update leaflet-ui-functions.js

- change the createMap() function so it targets a passed ID

alexandercyu committed:
Update leaflet-stac-ui.js

- create the leaflet map in the element with the passed ID

alexandercyu committed:
Update catalog-search.js

Capitalize text for Variable types

alexandercyu committed:
Update stac-ui.css

- Add styles for redesigned metadata section (as per meeting 2025-04-02)

alexandercyu committed:
Update catalog-search.js

- Redo functionality to output metadata (properties information) so that the properties output are now subsections under the general Metadata section
    - code for Metadata removed from under 'if(featureKey == "properties)' on line 772 and rewritten so it depends on 'stac_extensions' being a key in the features on line 478
- add BBox section

alexandercyu committed:
Update leaflet-stac-ui.js

Add option to not include widgets in leaflet map when loading a new map

alexandercyu committed:
Update leaflet-ui-functions.js

Add option to not include widgets in leaflet map when loading a new map

alexandercyu committed:
Update index.html

Add BBox template

alexandercyu committed:
Update catalog-search.js

- change condition so it adds different classes according to property type
- add classes so borders can be added  differently to different property types
- change how metadataValueCellDiv is added to the section body so it doesn't append an empty div in the Dimensions section

alexandercyu committed:
Update stac-ui.css

-Add styles for Dimension section

alexandercyu committed:
Update index.html

Remove initialization of map on page load

alexandercyu committed:
Update catalog-search.js

- add functionality to process geometry stac node if it exists
- move map creation for STAC search here and make it dependent on existence of geometry node
- add polygon to map if polygon coordinates exists

alexandercyu committed:
Update leaflet-stac-ui.js

- change initialization function for map to include parameter for polygon coordinates

alexandercyu committed:
Update leaflet-ui-functions.js

- change parameters of createMap() function to include parameter for polygon coordinates
- add function to add polygon from stac if it exists.

alexandercyu committed:
Update stac-ui.css

-Change style of dataset details headings so the chevron is at the end of the line

alexandercyu committed:
Update dropdown-panel.html

- Make default state of dropdown panel configurable through the jinja template settings
- wrap the magnifying glass icon and header text in a div so the spacing can be set separately from the chevron icon

alexandercyu committed:
Update index.html

Add default settings for dropdown panel (expanded or not expanded)

alexandercyu committed:
Update catalog-search.js

- change function to add polygon to map to a separate function call instead of using the createMap function
- change class added to dataset header

alexandercyu committed:
Update leaflet-stac-ui.js

- change map creation to instantiate map first and then create map

alexandercyu committed:
Update leaflet-ui-functions.js

- add global variable mapsPlaceholder to hold a map object that can be used by other functions outside of the createMap function context

alexandercyu committed:
Update leaflet-ui-functions.js

- Change initialization of map so it adds map object to an array to be used later to add STAC polygons to map

alexandercyu committed:
Update catalog-search.js

- Remove else condition when adding polygon to map
- add function to add bbox to map as a rectangle (need to format as array of coordinate pairs)

alexandercyu committed:
Update leaflet-ui-functions.js

- Add function to add map object to array for use by functions outside of createMap()
-  move cal to getShapeLatLng() to inside condition for addWidget boolean
- add addSTACBBox function to add bbox as rectangle

alexandercyu committed:
Update leaflet-ui.css

Expand width of map

alexandercyu committed:
Update stac-ui.css

- add styles for bbox and polygon legend for the map
- remove empty class

alexandercyu committed:
Update dropdown-panel.html

Add variable for expanding list or not by default

alexandercyu committed:
Update index.html

Add default values for showing dropdown list by default

alexandercyu committed:
Update catalog-search.js

Add functionality to display the bbox coordinates from the map in a list

alexandercyu committed:
Update leaflet-ui-functions.js

- Add a condition to the addSTACBBox function so it uses the correct entries when the incoming array has length 4 or length 6

- Add functionality to add a legend to the leaflet map when widgets are not shown (when map is only used for display purposes)

alexandercyu committed:
Update catalog-search.js

Remove console logs

alexandercyu committed:
Update stac-ui.css

- Add styles for legend
- remove unneeded class for legend

alexandercyu committed:
Update leaflet-ui.css

Add background colour to map container so it blends with the ocean colour

alexandercyu committed:
Update catalog-search.js

Metadata section now in expandable dropdown panel instead of static div

alexandercyu committed:
Update index.html

Add dropdown panel for metadata section

alexandercyu committed:
Update leaflet-ui-functions.js

Add tooltip labels with the lat and long for each corner in the bbox

alexandercyu committed:
Update stac-ui.css

Change alignment of left menu

alexandercyu committed:
Update main.html

Add ui-function-scripts.js call to main.html

alexandercyu committed:
Update dropdown.html

Remove unneeded script include

alexandercyu committed:
Update index.html

- Put search nav buttons and dataset introduction section into its own container div
- Remove unneeded container divs

alexandercyu committed:
Update catalog-search.js

- add code to show/hide the search nav / dataset introduction div
- fix bug where the dataset name was continually added and the previous dataset name was not erased
- fix bug where the bbox coordinates were continually added and the previous coordinates were not erased

alexandercyu committed:
Update ui-function-scripts.js

Add functionality to the Back button to hide the dataset details and show the search results

alexandercyu committed:
Update stac-ui.css

- Make containing div on right side take up full screen width to prevent the width adjusting to different widths of content

- Add padding for bbox coordinates
- Add background colour for Variables and Dimensions subtitles

alexandercyu committed:
Update catalog-search.js

- remove unneeded console statements
- add different classes for variables, dimensions, and other titles
- add class for bbox coordinate padding

alexandercyu committed:
Update catalog-search.js

-start filter function for dropdown on main search page
- remove unneeded console statements

alexandercyu committed:
Update index.html

- rename search display div ID
- run function to build out search display section

alexandercyu committed:
Update leaflet-ui-functions.js

- added function to check if range of longitude uses 0-360 or not
   -> if it does , return true.  If not, return false
- changed functionality of adding a STAC polygon/geometry to the leaflet map
   -> if the longitude uses a 0-360 range then convert that longitude to -180-180
    -> Use the converted coordinates to add the polygon to the map
- change functionality of addSTACBBox() function to check if any of the longitudes use 0-360 range
    -> if it does, convert it to use -180-180 range
- change text of legend to a more generic "Geometry"

alexandercyu committed:
Update catalog-search.js

- remove BBox section in the dataset details
- add call to longitudeRange360() function to check if the longitude uses 0-360 range
- add check to see if the longitude in the 'Dimensions' metadata uses 0-360 range
   -> if it does, convert it to -180-180 range
- IN PROGRESS: add function to build the "MySearch" area in the main search results
- IN PROGRESS: put testJSONString onto different lines just for readability for development

alexandercyu committed:
Update searchbox.html

Change magnifying glass icon to plus symbol

alexandercyu committed:
Update catalog-search.js

Add 'Geometry' header to page

alexandercyu committed:
Update checkbox.html

- change "properties" key to "summaries" to match change on backend

alexandercyu committed:
Update dropdown-default-scripts.js

- Add function to return STAC json so it can be used by other functions without multiple fetch functions
- expand function that builds the catalog filter to all checkboxes in the dropdown list
-  start functions to build filter for Frequency dropdown

alexandercyu committed:
Update dropdown-default-scripts.js

- add function to display what  is checked in the All Catalogs dropdown in the Datasets section in the My Search section

alexandercyu committed:
Update dropdown.html

- add function calls to functions that will :
   - select all checkboxes in the list when "All Catalogs" is checked
   - set the search page title to the appropriate catalog name when the checkbox is checked
   - display the catalog name(s) in the Dataset section when the checkbox is checked
- populate  the hidden div with the filter JSON created from checking the checkboxes

alexandercyu committed:
Update stac-ui.css

- add styles for My Search section

alexandercyu committed:
Update catalog-search.js

- test search results code and dataset details code with live data from redoak server (switchted testingURL to productionURL)

- fix error where the polygon data wasn't being added to map
- NOTE: leave commented code in place for areas where it is getting data from the redoak STAC JSON / testing server STAC JSON

alexandercyu committed:
Update index.html

- swap positions of dropdown source and dropdown share
- add advanced filter dropdowns

alexandercyu committed:
Update datepicker.html

- move datepicker function to its own file
- set the ending date to the  December 31 of following year from the current year

alexandercyu committed:
Update timepicker.html

- move timepicker function to its own file
-

alexandercyu committed:
Create datepicker.js

Date picker initialization moved to its own file

alexandercyu committed:
Create timepicker.js

Time picker initialization moved to its own file

alexandercyu committed:
Update stac-ui.css

Add margin to advanced filters as shown in design

alexandercyu committed:
Update button.html

-fix template name so it matches convention

alexandercyu committed:
Update dropdown-panel.html

- Make it so the time frame dropdown doesn't close when using flatpickr.
NOTE: this means the time frame dropdown needs to be closed manually

- set datepicker and timepicker IDs so they are customizable

alexandercyu committed:
Update catalog-search.js

- Convert timestamp to human readable date format

alexandercyu committed:
Update sample.html

Change template type name so it matches convention

alexandercyu committed:
Update index.html

- add class to style advanced filter dropdown section
- add time frame dropdown
- add Apply button to search filter area

alexandercyu committed:
Update datepicker.html

add function to set end year and end date for the datepicker to current end year and date

alexandercyu committed:
Update dropdown-panel.html

- add options for user set IDs for elements in time frame panel
- add function call to submit whatever is input in the timeframe to the Timeframe area in My Search

alexandercyu committed:
Update catalog-search.js

- Add condition to remove space in ID name

alexandercyu committed:
Update datepicker.js

- make flatpickr initialize on user supplied HTML element instead of fixed element
- add function to set end date and year of calendar to current end of year

alexandercyu committed:
Update dropdown-timeframe-scripts.js

Add function to add timeframe values into Time Frame section in My Search and format it for stac search
NOTE: not working

alexandercyu committed:
Update timepicker.js

Make it so timepickr initializes on user supplied HTML element instead of static element

alexandercyu committed:
Update index.html

Add settings for custom IDs for time frame panel elements

alexandercyu committed:
Update dropdown-timeframe-scripts.js

Datetime formatting works

alexandercyu committed:
Update leaflet-ui-functions.js

- removed conversion to -180-180 longitude range for bbox and polygon

alexandercyu committed:
Update catalog-search.js

- Remove test functions
- Remove calls to test functions

alexandercyu committed:
Update leaflet-ui-functions.js

- Remove coordinate check function

alexandercyu committed:
Update dropdown-panel.html

- add hidden div to hold the time range repeat dropdown value (NOTE: may not be used, but keep for now)
- make dropdown label  ID of the repeat dropdown user customizable
- add line to select All Day checkbox from DOM
- add function to All Day checkbox to disable the Start Time timepicker, End Date date picker, and End Time timepicker

alexandercyu committed:
Update dropdown.html

- fix dropdown list jinja variable so the ID for the list item (LI) is not just a number

alexandercyu committed:
Update index.html

- set the ID of the repeat dropdown label
- add div to display full filter string

alexandercyu committed:
Update catalog-search.js

- add functionality to the applyMySearch() function that will get the filter from the entered catalog and  time frame and add it to the existing unfinished filter string

alexandercyu committed:
Update dropdown-default-scripts.js

- change how catalog filter is built
   -> only build the argument for a selected catalog, not the entire filter

alexandercyu committed:
Update dropdown-timeframe-scripts.js

- get the timeframe filter working for a selected datetime range

alexandercyu committed:
Update catalog-search.js

- changed condition for checking dataset filter and timeframe filter so it checks if either filter has been entered
     -> prevents error when either filter has not been entered

- only add filter string to filter string array if a filter has been entered

alexandercyu committed:
Update dropdown-timeframe-scripts.js

- add "p" tags to display entered date range and time range
- clear time filter display before adding a new range
- format displayed times in UTC format
- add conditions to add a zero ("0") in front of the minute or second if it is less than 10
- format display time with colons between time values
-start function for building filter strings with repetition

alexandercyu committed:
Update catalog-search.js

- remove unneeded function
- change parent filter condition to 'and' because this function combines multiple filters
- no longer need to push filters from array items.  Just push the filter that is written to the hidden div

alexandercyu committed:
Update dropdown-default-scripts.js

- write the stringified JSON instead of an array

alexandercyu committed:
Update dropdown-timeframe-scripts.js

- add function to calculate days between dates for repetition queries
- add function to build filter string for repetition queries
- changed poulateTimeFrameFilter() function so it creates filter strings for 'datetime' parameter, 'end_datetime'. and 'start_datetime' parameters, time ranges
- change populateTimeFrameFilter() function so it will build the filter string even if only the end date is selected
- change populateTimeFrameFilter() function to fulfill the cases:
  -> only datetime is defined
  -> start_datetime and end_datetime are defined
  -> datetime and end_datetime are defined

alexandercyu committed:
Update dropdown-panel.html

change so only user defined ID is used for the ENds On datepicker

alexandercyu committed:
Update stac-ui.css

Add css for hiding the filters for each category

alexandercyu committed:
Update stac-ui.css

Add style to format display of human readable Start time and End time

alexandercyu committed:
Update dropdown-timeframe-scripts.js

- Fixed filter so 'start_datetime' uses 'datetime' to query
- Changed conditions for date and time inputs so if only the start date has a value the start time, end date, and end time are automatically populated to the same day as Start Date but the time is 24:00
(NOTE: flatpickr automatically converts dates at dd 24:00 to the next day, dd+1 00:00)
- added line to clear the hidden filter div for the submitted filter

alexandercyu committed:
Update catalog-search.js

-Fixed error with passed STAC url and the sortby clause where if the incoming url did not have a question mark ('?') the sortby clause would still be appended to the end of the url, causing an error when clicking 'Next'

alexandercyu committed:
Update dropdown-timeframe-scripts.js

Changed way start date and end date are displayed

alexandercyu committed:
Update dropdown-timeframe-scripts.js

Clean up

alexandercyu committed:
Update catalog-search.js

Removed unneeded console log

alexandercyu committed:
Update dropdown-timeframe-scripts.js

- remove unneeded code in calculateDays() function
- redo conditions for how time frame input is selected
- add code to build daily repetition filter string
- remove unneeded commented code

alexandercyu committed:
Update dropdown-timeframe-scripts.js

- Add repetitions for Daily, Weekly, Monthly
- need checks for Yearly and Decade
-TODO: Custom

mishaschwartz reviewed (changes requested):
I'm just going to focus on the date and time filters for this review. 

- If you're selecting a single range (no repeats) the query should differentiate between 
    1. an item with a single `datetime` (hint: there is no `end_datetime`)
    2. an item with a `datetime` and `end_datetime` (hint: there is no `start_time`)
    3. an item with a `start_datetime` and `end_datetime`
  
  Right now the query applies the rule that should only apply to 1. to both 1. and 2. ; and 3. is not handled.

- If you're selecting a single range (no repeats) the part of the query that is for selecting items with only a `datetime`  is currently selecting everything (hint: use 'and' not 'or').
- if "All day" is selected, the query is malformed
- if "All day" is not selected, the user should have to supply an end date
- there is no way to clear the selection from a time input
- Repeats are not handled
- "Selected Time Range" should be "Repeats" or "Repeating" (I know figma says otherwise but the current design is confusing)
- The "Never" option should be part of the "Ends On" selector. Otherwise its unclear what "Never" refers to (I know figma says otherwise but the current design is confusing)

I haven't looked at the code itself yet. I will review that once the comments above have been resolved. 

Please ask questions if you're not sure what to do.

alexandercyu committed:
Update catalog-search.js

Fixed issue where filtered search results will populate the Next and Previous buttons with urls that don't have the next/previous token included with the url

alexandercyu committed:
Update stac-ui.css

- added styles for time frame panel reset and review buttons
- added styles for time frame error messages

alexandercyu committed:
Update dropdown-panel.html

- Changed label for 'Never' radio button to 'Never Ends'
- Changed label for 'Selected Time Range' to 'Selected Time Range Repeats:'
- Added a Reset button at the bottom of time frame panel
- added functions for Reset button to reset the inputs in the timeframe panel
- added function on the submit/Review button so it closes the time frame panel on click

alexandercyu committed:
Update dropdown-panel.html

- change the  jinja'dropdown_list_id' to 'dropdown_timeframe_list_id' for the timeframe template

alexandercyu committed:
Update index.html

- change the  jinja'dropdown_list_id' to 'dropdown_timeframe_list_id' for the timeframe template

alexandercyu committed:
Update catalog-search.js

- remove unneeded consoles
- remove unneeded commented code

alexandercyu committed:
Update dropdown-list.js

- add attribute 'swappedID' so the list item can be accessed later to reset the timeframe panel

alexandercyu commented:
> I'm just going to focus on the date and time filters for this review.
> 
>     * If you're selecting a single range (no repeats) the query should differentiate between
>       
>       1. an item with a single `datetime` (hint: there is no `end_datetime`)
>       2. an item with a `datetime` and `end_datetime` (hint: there is no `start_time`)
>       3. an item with a `start_datetime` and `end_datetime`
>       
>       Right now the query applies the rule that should only apply to 1. to both 1. and 2. ; and 3. is not handled.
> 
>     * If you're selecting a single range (no repeats) the part of the query that is for selecting items with only a `datetime`  is currently selecting everything (hint: use 'and' not 'or').
> 
>     * if "All day" is selected, the query is malformed
> 
>     * if "All day" is not selected, the user should have to supply an end date
> 
>     * there is no way to clear the selection from a time input
> 
>     * Repeats are not handled
> 
>     * "Selected Time Range" should be "Repeats" or "Repeating" (I know figma says otherwise but the current design is confusing)
> 
>     * The "Never" option should be part of the "Ends On" selector. Otherwise its unclear what "Never" refers to (I know figma says otherwise but the current design is confusing)
> 
> 
> I haven't looked at the code itself yet. I will review that once the comments above have been resolved.
> 
> Please ask questions if you're not sure what to do.

- 'datetime' query changed to an 'and' query, but not sure if it's the correct use 

- 'start_datetime' - 'end_datetime' query added for non-repeating time ranges (This generates the same serve return error as before.  I interpreted your solution to mean 'datetime'  could be used in place of 'start_datetime' in a 'start_datetime' - 'end_datetime' query)
- 
- Added repeats, but they are only for 'datetime' - 'end_datetime' queries.  Working on 'start_datetime' - 'end_datetime' queries
- added a Reset button to clear the inputs
- Fixed the 'All Day' query
- if 'All Day' is checked the Start Time, End Time, and End Date are cleared and disabled.  Start Time and End Time are automatically made'00:00' and '24:00'. Timepickr automatically converts the End Date and End Time to morning of the next day (Eg. May 5 24:00 becomes May 6 00:00)
- Changed  the 'Selected Time Range' text to 'Selected Time Range Repeats:'
- Changed the 'Never' text to 'Never Ends'

alexandercyu committed:
Update dropdown-timeframe-scripts.js

- Added the ability for the checkboxToggleAllDay function to clear and disable the end date input

alexandercyu committed:
Update dropdown-panel.html

- Added the ability for the checkboxToggleAllDay function to clear the end date input

alexandercyu committed:
Update stac-ui.css

- Removed margin for Ends On radio button
- Lined up Ends On elements so they align with other time frame elements at the start of the line and end of the line

alexandercyu commented:
> Added repeats, but they are only for 'datetime' - 'end_datetime' queries. Working on 'start_datetime' - 'end_datetime' queries

Added repeats with 'start_datetime' - 'end_datetime' queries

alexandercyu committed:
Update dropdown-panel.html

- Changed text of 'Never Ends' radio to 'Repeat Never Ends'
- Changed text of 'Ends On' radio to 'Repeat Ends On'
- enabled 'populateStacVersion()' function

alexandercyu committed:
Update dropdown-timeframe-scripts.js

- Added building of 'start_datetime' - 'end_datetime' repeat queries
- removed unneeded commented code
- removed empty lines

alexandercyu committed:
Update dropdown-timeframe-scripts.js

Removed unneeded commented code

alexandercyu committed:
Update stac-ui.css

- Fix styles for left menu for timeframe item
- Fix styles for left menu for alignment of chevrons and button panel text
- Add styles for Home/Browse button
- Add padding for right content area

alexandercyu committed:
Create open-in-app.svg

Add icon for Home/Browse button

alexandercyu committed:
Update button-radio.html

Add label text class

alexandercyu committed:
Update datepicker.html

- Change label text class
Change format of placeholder text

alexandercyu committed:
Update dropdown-panel.html

- add text class for regular_dropdown
- add text class for radio buttons
- change label for datepicker
- comment out templates for repeat time frame UI elements
- comment out scripts for repeat time frame UI elements that load on page load

alexandercyu committed:
Update dropdown.html

- added different options for title styles

alexandercyu committed:
Update index.html

- added elements to Home/Browse button
- ADded Advanced FIlters title

alexandercyu committed:
Update datepicker.js

Enabled date selector and time selector in one widget

alexandercyu committed:
Update timepicker.js

Made time format 24 hour

alexandercyu committed:
Update stac-ui.css

Remove underline in back button

alexandercyu committed:
Update index.html

Add text styles to back button text

alexandercyu committed:
Update catalog-search.js

Add function to swap back button text to dataset link

alexandercyu committed:
Update ui-function-scripts.js

Add function to swap back button text

alexandercyu committed:
Update ui-styles-scripts.js

Create function to swap back button text

alexandercyu committed:
Update stac-ui.css

Add margins to search description area

alexandercyu committed:
Update index.html

Add class for margins in search description area

alexandercyu committed:
Create banner-montreal.png

Add new banner image

alexandercyu committed:
Update banner.html

-Remove banner video
- Add banner image

alexandercyu committed:
Update stac-ui.css

-update styles for searchbox and default dropdown so they align center to the page
- change banner video to banner image

alexandercyu committed:
Update banner.html

- remove banner video, add banner image for large screen sizes
- add class for aligning dropdown in search bar

alexandercyu committed:
Update dropdown.html

- add class for aligning dropdown in search bar

alexandercyu committed:
Update index.html

- add class for background image for large screen size
- set ID for Frequency dropdown
- add function call to build Frequency list items

alexandercyu committed:
Update catalog-search.js

- add functions to add Frequency dropdown list items from stac queryables or stac collections

alexandercyu committed:
Update index.html

- add class to hide full filter div

alexandercyu committed:
Update catalog-search.js

- move unneeded repeat code to its own file
- add hidden div class to hidden filter divs

alexandercyu committed:
Update stac-ui.css

- change large banner image because the sun in the previous image was blocking some of the title text
- override bootstrap so the dropdown panels aren't pushed down

alexandercyu committed:
Create banner-forest.png

New banner large image

alexandercyu committed:
Update stac-ui.css

-add styles for search-input-container
- switch hover and focus styles to the search input container
- position search input textbox so it's just after the All Catalogs dropdown
- left align input text so it starts just after the All Catalogs dropdown and so input text doesn't over flow on the left (until it reaches the limit on the right)

alexandercyu committed:
Update searchbox.html

Make the search input container change its borders on inputting text instead of the input field
- move style class and aria-expanded attribute to the search input container

alexandercyu committed:
Update catalog-search.js

- Switch the aria-expanded attribute on the search input container instead of on the search input itself

alexandercyu committed:
Update search-ui-functions.js

Add ID for search input container for getWord() function to use

alexandercyu committed:
Update searchbox.html

Remove call to setInputTextStart() function

alexandercyu committed:
Update catalog-search.js

- Add  row to search results table to display error message if no results returned
- add Frequency filter to the function that builds the full filter string
- make function that creates the filter string for each Frequency

alexandercyu committed:
Update stac-ui.css

- Add styles for advanced filter list item buttons
- change how hidden divs for filter strings are hidden
- add error style for search result errors

alexandercyu committed:
Update leaflet-ui.css

Add styles to display leaflet map in a modal

alexandercyu committed:
Update stac-ui.css

- Change styles for 'X' close button for location modal on stac search page
- Add styles for location button

alexandercyu committed:
Update button.html

Add template for Location panel button

alexandercyu committed:
Create modal-map.html

Add modal partial for leaflet map inside a modal

alexandercyu committed:
Update index.html

- Remove list of hardcoded item labels for Frequency dropdown
- Add Location button
- Add map modal

alexandercyu committed:
Update catalog-search.js

- Modify buildFrequencyFilterDropdown() function so the part that populates the dropdown is separated into its own function.
- Modify buildFrequencyFilterDropdown() function so it properly processes Collections json
- Add functionality to write the frequency chosen in the MySearch area

alexandercyu committed:
Update dropdown-default-scripts.js

- Remove unneeded functions for building Frequency dropdown

alexandercyu committed:
Update leaflet-stac-ui.js

- Add functions to show modal with Leaflet map in it
- Move the map instantiation for the modal inside the button click so it loads the map after the modal is shown (fixes error where map loads background images according to size of containing div)

alexandercyu committed:
Update leaflet-ui-functions.js

- comment out icons for Leaflet map button that converts drawn shapes to geojson
--> use text instead: "Review"
- comment out anything that uses divLatLng and clearText.  This was for showing the coordinates to the user before the search UI was finalized
- add functionality to display the coordinates of the drawn shape in the search filter area

alexandercyu committed:
Update leaflet-ui.css

Added cursor style to Review button

alexandercyu committed:
Update stac-ui.css

- added style to Location button so background changes to white on hover
- added style so each filter in My Search has a max height and scrolls when that height is exceeded

alexandercyu committed:
Update catalog-search.js

- Add css class to style My Search filter sections
- add Frequency filter and Location filter to the filter to be submitted to the server
- remove testing code and commented code

alexandercyu committed:
Update dropdown-timeframe-scripts.js

Remove code for UI elements for repeated sections (these are moved to another file)

alexandercyu committed:
Update leaflet-ui-functions.js

- add function to build filter for shapes drawn on map
- NOTE: keep commented lines with different intersect queries

alexandercyu committed:
Update search-ui-functions.js

Move spin button functions run on load to spinbutton.js

alexandercyu committed:
Update spinbutton.js

Move spin button functions run on load to spinbutton.js from search-ui-functions.js

alexandercyu committed:
Update catalog-search.js

-create new div to hold keyword tags
- add ID for keyword div

alexandercyu committed:
Update stac-ui.css

- add style for keyword tag container
- change styles for dropdown buttons so they don't change colour on hover

alexandercyu committed:
Update searchbox.html

- include scripts to add keyword tags

alexandercyu committed:
Update tag.html

add keyword tag template

alexandercyu committed:
Update leaflet-ui-functions.js

Add check to make new map ID null if there is already a map initialized in the map container.
-> prevents error when there is already a map initialized in the map container
- Add TODo comments for intersects filters for map coordinates

alexandercyu committed:
Update search-ui-functions.js

Add function to add keyword tag on click of plus (+) button in search box

alexandercyu committed:
Create tag-scripts.js

- add scritps to add keyword tag, remove
- TODO: build filter with keywords

alexandercyu committed:
Update index.html

Moved Timeframe and Location buttons to the top of the list

alexandercyu committed:
Update leaflet-ui.css

Removed scrollbars on map modal when screen shrinks smaller than map

alexandercyu committed:
Update leaflet-ui.css

Change dimensions of map containers map shrinks and grows with screen

alexandercyu committed:
Update catalog-search.js

- Add 'Other' category for My Search area to display queryables filter
- add in the display divs and hidden divs for the 'OTher' category
- add display divs and hidden divs for the searchbox keywords
- remove unused variable 'argumentArray'
- add function to build additional dropdowns in the Advanced Filter area from the queryables in the queryables endpoint
- add function to build the queryables filter string based on the checkboxes the user selects from the queryables advanced filters
-> will also dynamically update the Other display area to remove items when checkbox is unchecked.

alexandercyu committed:
Update index.html

- add menu button to open map
- add div to hold queryables dropdowns
- add function call to build queryables dropdowns and filter

alexandercyu committed:
Update leaflet-ui-functions.js

- change map filter to use Intersect instead of s_intersects

alexandercyu committed:
Update modal-map.html

Change button colour for the close button in the map modal

alexandercyu committed:
Update dropdown.html

- add entry for frequency dropdown
- remove unneeded code

alexandercyu committed:
Update button.html

- add button colour class for styling

alexandercyu committed:
Update stac-ui.css

- add classes for styling close button in map modal
- add scrolling in frequency dropdown if it needs it
- add styles for frequency dropdown for hover and spacing
- add styles for keyword tags for spacing

alexandercyu committed:
Update index.html

Add Clear button

alexandercyu committed:
Update catalog-search.js

- remove unused function
- Add Clear button
- align for loop

alexandercyu committed:
Update searchbox.html

Add bootstrap attribute for closing the keyword list when clicking outside the list

alexandercyu committed:
Update catalog-search.js

- add the stac property key to the keyword return (eg. cmip6:variable_id)
- add the stac property key and the queryable value on the attributes of the DOM element (keyword list item, input box when list item is clicked)
- remove commented code
- remove uneeded TODO lines
- add hidden div for keyword tag filter

alexandercyu committed:
Update dropdown-timeframe-scripts.js

- move code to timeframe-repeat.js
- delete unused variables
- delete commented code
- add function to calculate days between dates
- remove unneeded console log statements

alexandercyu committed:
Update search-ui-functions.js

- add functionality to get stac property key and queryable property to pass on to addKeywordTag function

alexandercyu committed:
Update tag-scripts.js

- modify addKeywordTag() function to add the stac property key and queryable value to the global keyword list
- modify removeKeywordTag() function to delete the keyword entry from the global list and the hidden filter div
- add functionality to build the keyword filter and add it to the hidden keyword div
- add styles for close button on tag

alexandercyu committed:
Update catalog-search.js

- add keyword filter to the whole filter sent to the server when clicking Apply

alexandercyu committed:
Update ui-styles-scripts.js

Added comment for unused function
NOTE: check where function should be used and move to other file if needed

alexandercyu committed:
Update banner.html

Change 'Catalog' to 'Collections'

alexandercyu committed:
Update checkbox.html

Remove unneeded commented code

alexandercyu committed:
Update dropdown.html

Change 'Catalog' to 'Collections'

alexandercyu committed:
Update searchbox.html

- change target of aria-controls to the list instead of the container
- remove reference to tag-scripts.js

alexandercyu committed:
Update index.html

- remove 'Select'

alexandercyu committed:
Update catalog-search.js

- Change 'Datasets' to 'Collections'
- Remove 'Sub-Categories' and 'Format'
- remove keyword tag container div element
- remove 'Select' from dropdown buttons
- add function to check if duplicate item has been added to Other.  Remove it if there is a duplicate and add the new item
- copy functions from tag-scripts.js into here

alexandercyu committed:
Update dropdown-default-scripts.js

- Change 'Catalogs' to 'Collections'
- Change 'Datasets' to 'Collections'

alexandercyu committed:
Update search-ui-functions.js

Remove 'Tag' from function name

alexandercyu committed:
Delete tag-scripts.js

File no longer needed

alexandercyu committed:
Update stac-ui.css

- Removed styles preventing checkbox label text word wrap
- added styles to checkbox label
- added padding the dropdown list item
- changed dimensions of checkmark

alexandercyu committed:
Update catalog-search.js

- grouped Advanced Filter dropdowns by extension name and extension type
- removed or renamed references to searchFilterKeywordHidden elements
- functions for building Advanced Filter dropdowns changed to use curated fuse dictionary list instead of  raw queryables list from server

alexandercyu committed:
Update index.html

- changed function call to build Advanced Filter list

alexandercyu committed:
Update catalog-search.js

- add command to specifically remove quote chracter from around the checkbox ID
- fixed issue where a checkbox value would be displayed multiple times if multiple checkboxes were selected
- remove unneeded functions from when the keyword tag feature was implemented
- merge addKeyword and build keyword filter functionality with similar functions triggered by the checkboxes

alexandercyu committed:
Update catalog-search.js

- Clear button now clears checkboxes on left side

alexandercyu committed:
Update catalog-search.js

- added functionality to clear checkboxes in the searchbox dropdown if they are checked
- added functionality so only queryables with an enum entry are visible in the left checkboxes.

alexandercyu committed:
Update checkbox.html

Removed extra end div

alexandercyu committed:
Update search-ui-functions.js

Remove queryableValue variable from parameter since it's not being used in the addKeyword function

alexandercyu committed:
Update leaflet-ui.css

Fix styling of map search panel

alexandercyu committed:
Update stac-ui.css

- fix alignment of searchbox and dropdown in search bar
- fix hover colours of search bar

alexandercyu committed:
Update stac-ui.css

Add padding to search 'plus' icon

alexandercyu committed:
Update banner.html

fix alignment of search bar elements

alexandercyu committed:
Update searchbox.html

ONGOING
- find out why dropdown list does not collapse when clicking out of it

alexandercyu committed:
Update catalog-search.js

- remove underscore from queryable dropdown names
- remove unneeded variables from addKeyword() function
- add condition to check for null or empty string values in addKeyword() function

alexandercyu committed:
Update leaflet-ui-functions.js

Remove testing console log

alexandercyu committed:
Update index.html

- Add div for non-enum entries from STAC
- remove call for buildFrequencyFilterDropdown function
- Add TODO to clean up frequency dropdown function calls and ID use

alexandercyu committed:
Update catalog-search.js

- add conditions to populate queryables that don't have a prefix/extension name but still have an enum entry into the dropdowns on the left side
   -> will be put under Location
- add conditions to remove underscore from queryable name and format as needed
- add conditions to format 'id' and 'url' in the queryable name as all caps
- remove the previous methods of styling queryable names with underscores
- add conditions  in the buildQueryables() function to populate the 'Frequency' queryables under the 'Frequency' area instead of the 'Other' area
- add conditions in the buildQueryables() function to remove items added under the 'Frequency' area

alexandercyu committed:
Update index.html

Comment out the Frequency dropdown template for cleanup

alexandercyu committed:
Update dropdown-default-scripts.js

change name of function to be less generic

alexandercyu committed:
Update dropdown.html

change name of function to be less generic

alexandercyu committed:
Update checkbox.html

change name of function to be less generic

alexandercyu committed:
Update catalog-search.js

Added call to function to reset search bar dropdown to default title text

alexandercyu committed:
Update index.html

Remove Frequency dropdown template

alexandercyu committed:
Update catalog-search.js

Remove unneeded functions used to build Frequency dropdown

alexandercyu committed:
Update leaflet-ui-functions.js

Remove unused function and function calls

alexandercyu committed:
Update ui-function-scripts.js

Remove unused function

alexandercyu committed:
Update ui-styles-scripts.js

Comment out spin button function for now and leave explanation comment

alexandercyu committed:
Update leaflet-ui-functions.js

-Rename getShapeLatLng() function to getMarkerLatLng()
- remove unneeded functionality
  -> functionality done by 'Review' button / taken over by formatGeoJSON() function

alexandercyu committed:
Update leaflet-stac-ui.js

Use existing function to set modal

alexandercyu committed:
Update leaflet-ui-functions.js

Comment out solution for error "Map container is already initialized." for now.
This solution interferes with coordinates being added to Point field

alexandercyu committed:
Update leaflet-ui-functions.js

Put variable for 'Map container is already initialized.' solution inside commented area

alexandercyu committed:
Create timeframe-repeat.js

Functions for repeated time frames separated out of main code and saved for later when issues with faking repetition query filters are resolved

alexandercyu committed:
Update catalog-search.js

Add functionality to close the keyword search result list when clicking the list item

alexandercyu committed:
Update checkbox.html

Checkbox width  changes with text width

alexandercyu committed:
Update catalog-search.js

Update functionality to build filter dropdowns on left side to include ability to build dropdowns from the /collections endpoint  summaries as well.

alexandercyu committed:
Update catalog-search.js

Remove unneeded functions (grouping of collections and queryables endpoint json handled by one function now)

alexandercyu committed:
Update index.html

Remove call to old build queryables filter dropdown function

alexandercyu committed:
Update dropdown-panel.html

remove comment

mishaschwartz commented on static/css/stac-ui.css:
```diff
@@ -337,6 +346,155 @@ h6{
+    z-index: -99;
+}
+
+.banner-image-node{
```
unused

mishaschwartz commented on static/css/stac-ui.css:
```diff
@@ -337,6 +346,155 @@ h6{
+    z-index: -99;
+}
+
+.banner-image-about{
```
unused

mishaschwartz commented on static/css/stac-ui.css:
```diff
@@ -337,6 +346,155 @@ h6{
+    z-index: -99;
+}
+
+#banner-video{
```
unused

mishaschwartz commented on templates/site/index.html:
```diff
@@ -0,0 +1,209 @@
+        {% set banner_background_class= "banner-image-home" %}
+        {% set banner_large_background_class = "banner-image-home-large" %}
+        {% set banner_title_class = "banner-title-home" %}
+        {% set banner_video = "video/banner-video.mp4" %}
```
unused

mishaschwartz commented on templates/site/index.html:
```diff
@@ -0,0 +1,209 @@
+        {% set banner_large_background_class = "banner-image-home-large" %}
+        {% set banner_title_class = "banner-title-home" %}
+        {% set banner_video = "video/banner-video.mp4" %}
+        {% set page_title = "Data Analytics for Canadian Climate Services, STAC API" %}
```
These should all come from the STAC catalog. Not hardcoded

mishaschwartz commented on templates/partials/banner.html:
```diff
@@ -0,0 +1,79 @@
+
+                <div class="row">
+                    <div class="container-fluid">
+                        <div class="banner-search-container padding-unset">
```
Should be centred horizontally

mishaschwartz replied:
![image](https://github.com/user-attachments/assets/8fb5adcd-698f-4af5-92cf-3098915a5cee)

This is not fixed inside the banner and it's height relative to the banner changes significantly depending on the banner size, leading to the sort of thing in the picture above

mishaschwartz commented on templates/partials/banner.html:
```diff
@@ -0,0 +1,79 @@
+        <div class="banner-overlay-container">
+            <div class="container-fluid">
+                <div class="row">
+                    <div class="d-flex justify-content-center">
```
All text here should be centre aligned

mishaschwartz commented on templates/partials/banner.html:
```diff
@@ -0,0 +1,79 @@
+            </div>
+        </div>
+    {% else %}
+        {% if page == "node" %}
```
page is only ever "home" in this code so this and everything below it are never used

mishaschwartz commented on static/images/banner/banner-montreal.png:
unused

mishaschwartz commented on static/video/banner-video.mp4:
unused

mishaschwartz commented on templates/site/index.html:
```diff
@@ -0,0 +1,209 @@
+    <div class="container-fluid content-search-main">
+        <div class="content-search-left">
+        <div id="backButtonContainer" class="div-back-button-container">
+            <i class="fa-solid fa-arrow-left back-button-arrow"></i><a id="backButton" class="button-back-button subtitle-1" role="button">Home</a><img  src="images/open-in-app.svg">
```
This button shows up on the search page when the page first loads. I don't know what Home means in this context and if you click it, it takes you to an empty dataset page with no info and a partial map

mishaschwartz commented on templates/site/index.html:
```diff
@@ -0,0 +1,209 @@
+                            <div id="datasetGeneralContainer"></div>
+
+                            <div id="datasetGeometryContainer">
+                                <div id="datasetMapContainer" class="map"></div>
```
The map doesn't render properly here:

![image](https://github.com/user-attachments/assets/b5a4e082-b3df-4cf3-a60f-3f84f4411958)

mishaschwartz replied:
Header gets added multiple times if you choose different items without reloading the page:

![image](https://github.com/user-attachments/assets/6d3b69e5-9fd7-4b5e-bb36-199d896bfc82)

mishaschwartz commented on templates/site/index.html:
```diff
@@ -0,0 +1,209 @@
+                    <div id="searchResults"></div>
+                    <div id="datasetDetails">
+                        <div id="datasetContainer" class="div-container-dataset">
+                            <div id="datasetCollectionContainer">
```
Text style does not match Figma

![image](https://github.com/user-attachments/assets/b012c90e-e3af-460c-b7ae-adeef5ec859a)

mishaschwartz commented on templates/site/js/catalog-search.js:
```diff
@@ -143,3 +157,1170 @@ function addDefaultSearchAttributes(inputBox){
+            if (featureValue.properties.start_datetime) {
+                var cellStartDateTimeValue = document.createElement("td");
+
+                cellStartDateTimeValue.innerText = new Date(featureValue.properties.start_datetime);
```
actually format the date, don't just rely on the default string representation because then you get inconsistent timezone formats:

![image](https://github.com/user-attachments/assets/a0026da1-0f24-4062-aad3-1a353c4e00ae)

mishaschwartz commented on templates/site/index.html:
```diff
@@ -0,0 +1,209 @@
+                        sensor type, let you further customize your results to meet quality and relevance standards.
+                    </p>
+                </div>
+                <div id="searchFilterContainer" class="div-search-filter-container">
```
This does not match the design. This should be in a box and the buttons should be formatted differently.

mishaschwartz commented on templates/site/index.html:
```diff
@@ -0,0 +1,209 @@
+                        </div>
+                </div>
+
+                <div id="searchResultsNavigation" class="search-results-nav">
```
Does not match the design in too many ways to list them... just go look at Figma again please.

mishaschwartz commented on templates/site/index.html:
```diff
@@ -0,0 +1,209 @@
+
+            <div id="nonEnumQueryables" class="div-queryable-filters"></div>
+
+            <div id="dropdownQueryables" class="div-queryable-filters"></div>
```
Why are some labels left aligned and some centred?
Checkbox should be vertically aligned to the centre of the label. 

![image](https://github.com/user-attachments/assets/7ef7c588-9648-4542-93d0-9eac234aa022)

alexandercyu replied:
This is a mystery to me as well because they all have the same text styling classes.

mishaschwartz commented on templates/site/index.html:
```diff
@@ -0,0 +1,209 @@
+                <h6 class="title-advanced-filters">Advanced Filters</h6>
+            </div>
+
+            {% with %}
```
Why does the time frame dropdown have a box shadow but none of the others do?

![image](https://github.com/user-attachments/assets/c5911be6-dc52-4e41-be28-b47c72692be9)

mishaschwartz reviewed (changes requested):
This is an initial review that is only looking at the style and how things are presented on the page. I will do another review later looking at the code itself.

The main takeaways are:

- The style is not consistent with what is presented in Figma. I know that we chose to add/remove some elements but the styling should be the same
- the map component needs work, by fixing it in the modal you broke it on the dataset page... it needs to work in both cases.
